dftimewolf/lib/exporters/s3_to_gcs.py

```diff
@@ -0,0 +1,155 @@
+        files are from a previous module but not all should be transferred.
+    """
+    self.aws_region = aws_region
+    self.dest_project_name = dest_project
+    self.dest_bucket = dest_bucket
+    self.dest_project = gcp_project.GoogleCloudProject(self.dest_project_name)
+    self.filter = object_filter
+
+    if s3_objects:
+      for obj in s3_objects.split(','):
+        self.state.StoreContainer(containers.AWSS3Object(obj))
+
+  def PreProcess(self) -> None:
+    """Prep work for copying objects from S3 to GCS - Bucket creation."""
+    # Check if the destination bucket exists. If not, create it.
+    if not self.dest_bucket:
+      self.ModuleError('No destination GCP bucket specified', critical=True)
+    if self.dest_bucket not in \
+        [bucket['id'] for bucket in self.dest_project.storage.ListBuckets()]:
+      self.logger.info('Creating GCS bucket {0:s}'.format(self.dest_bucket))
+      self.dest_project.storage.CreateBucket(self.dest_bucket)
+
+    # Set the permissions on the bucket
+    self.logger.info('Applying permissions to bucket')
+    self._SetBucketServiceAccountPermissions()
+
+  def Process(self, container: containers.AWSS3Object) -> None:
+    """Creates and exports disk image to the output bucket."""
+    if self.filter:
+      if not re.match(self.filter, container.path):
+        self.logger.info('{0:s} does not match filter. Skipping.'\
+            .format(container.path))
+        return
+
+    # We must create a new client for each thread, rather than use the class
+    # member self.dest_project due to an underlying thread safety issue in
+    # httplib2: https://github.com/googleapis/google-cloud-python/issues/3501
+    client = gcp_project.GoogleCloudProject(self.dest_project_name)
+    client.storagetransfer.S3ToGCS(
+        container.path, self.aws_region + 'a', 'gs://' + self.dest_bucket)
+    _, s3_path = SplitStoragePath(container.path)
+    output = self.dest_bucket + '/' + s3_path
+    self.state.StoreContainer(containers.GCSObject(output))
+
+  def _SetBucketServiceAccountPermissions(self) -> None:
+    """Grant access to the storage transfer service account to use the bucket.
+    See https://cloud.google.com/storage-transfer/docs/configure-access#sink"""
+
+    request = self.dest_project.storagetransfer.GcstApi().\
+        googleServiceAccounts().get(projectId=self.dest_project_name)
+    service_account = request.execute()['accountEmail']
+
+    client = storage_client(project=self.dest_project_name)
+    bucket = client.get_bucket(self.dest_bucket)
+    policy = bucket.get_iam_policy(requested_policy_version=3)
+    policy.version = 3
+
+    policy.bindings.append({
+        'role': 'roles/storage.legacyBucketWriter',
+        'members': ['serviceAccount:' + service_account],
+    })
+    policy.bindings.append({
+        'role': 'roles/storage.objectViewer',
+        'members': ['serviceAccount:' + service_account],
+    })
+
+    bucket.set_iam_policy(policy)
```

Comment:
Do we need to do any error handling or exception management for any of these calls?